algviz/interface/visitors.py

```diff
@@ -190,3 +190,29 @@ class WidgetVisitor(Visitor):
 
     def visit(self, *args, **kwargs):
         return super().visit(*args, **kwargs)
+
+class LinkedListVisitor(Visitor):
+    """Linked lists are represented as an array of an array and a current index, i.e.,
+    [[1, 2, 3], 0]."""
+    type_ = Tokens.ARRAY_T#treat this as an array, at least for now.
+
+    def has_current(self, ll):
```

Comment:
Instead of `has_current` and `has_next`, let's have a single method called `is_nil` that says whether the given list element is the empty list.
I think you were trying to account for people who throw an exception if the equivalent of `get_next` is called on the last element of their list.  But such a person can just do:
```python
def get_next(self, ll):
    try:
        return ll.successor()  # or whatever
    except MyEndOfListError:
        return None
def is_nil(self, ll):
    return ll is None
```